v2/tests/log_context_property_type_ascii_char_ptr_ut/log_context_property_type_ascii_char_ptr_ut.c

```diff
@@ -0,0 +1,530 @@
+
+    // assert
+    POOR_MANS_ASSERT(result == 3);
+    POOR_MANS_ASSERT(strcmp(buffer, "bau") == 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_005: [ If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. ]*/
+static void when_snprintf_fails_ascii_char_ptr_to_string_also_fails(void)
+{
+    // arrange
+    char buffer[4];
+    setup_mocks();
+    expected_calls[0].mock_call_type = MOCK_CALL_TYPE_snprintf;
+    expected_calls[0].u.snprintf_call.override_result = true;
+    expected_calls[0].u.snprintf_call.call_result = -1;
+    expected_call_count = 1;
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string((void*)"bau", buffer, sizeof(buffer));
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy */
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_006: [ If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall fail and return a non-zero value. ]*/
+static void ascii_char_ptr_copy_with_NULL_src_value_fails(void)
+{
+    // arrange
+    char buffer[128];
+    setup_mocks();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy(buffer, NULL);
+
+    // assert
+    POOR_MANS_ASSERT(result != 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_007: [ If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall fail and return a non-zero value. ]*/
+static void ascii_char_ptr_copy_with_NULL_dst_value_fails(void)
+{
+    // arrange
+    char buffer[128];
+    setup_mocks();
+    setup_expected_vsprintf_call();
+    LOG_CONTEXT_PROPERTY_TYPE_INIT(ascii_char_ptr)(buffer, "cucu");
+    setup_mocks();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy(NULL, buffer);
+
+    // assert
+    POOR_MANS_ASSERT(result != 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_008: [ `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall copy the entire string (including the `NULL` terminator) from `src_value` to `dst_value`. ]*/
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_009: [ `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall succeed and return 0. ]*/
+static void ascii_char_ptr_copy_succeeds(void)
+{
+    // arrange
+    char src_buffer[128];
+    char dst_buffer[128];
```

Comment:
>dst_buffer[128]; [](http://example.com/codeflow?start=9&length=16)

this is too generous here, let's make it 3!

Reply:
Since we say that the caller has to ensure there is enough memory I can't really write that test.